nservicebus/outbox/index.md

```diff
@@ -79,8 +79,61 @@ In order to gradually convert an entire system from the DTC to the Outbox:
+
+
+## Non transactional resources
+
+If non-transactional resources are being accessed then make sure processing is idempotent. Non-transactional resources are not guaranteed to be handled in an idempotent way by the Outbox. If more-than-once invocation is unwanted then this can be almost prevented completely by making sure only a single endpoint instance is running and have [configured the endpoint to process messages sequentially by setting the maximum concurrency to 1](/nservicebus/operations/tuning.md#tuning-concurrency). Only data modified via a shared storage session is made idempotent via the outbox and has exact-once semantics. See [persistence specific documention](/nservicebus/outbox/#Persistence) on how to obtain the shared storage session.
+
+NOTE: This problem is not specific to Outbox but mentioned here as sometimes it is assumed Outbox does deduplication that prevents more-than-once invocation.
+
+Try to avoid mixing transactional and non transactional tasks. If performing non transactional tasks send a message to perform this task in isolation.
+
+
+## Outbox expiration duration
+
+Part of the outbox is to guarantee that the data remains consistent if the same message is processed more-than-once. A message with the exact same identification can be detected and ignored. This data is stored in the outbox record and kept for a configurable duration. The default expiration duration depends per persistence implementation. All implemetations can be configured on how long an outbox record needs to be stored and some can be configured on how often outbox cleanup needs to occur.
+
+Ensure that the retention period of the outbox is longer than the maximum time the message can be retried, including delayed retries and manual retries via ServiceControl. Additional care must be taken by operators of ServicePulse to not retry messages older than the outbox retention period.
+
+## Required storage
+
+Outbox requires storage. The required space can be calculated as follows:
+
+    Total outbox records = Message througput per second * Deduplication period in seconds
+
+A single outbox record - after all transport operations have been dispatched - usually requires less than 50 bytes of which most are taken for storing the original message ID as this is a string value.
+
+NOTE: If the system is processing a high volume of messages per second having a large deduplication timeframe might not be ideal as outbox records will occupy storage.
+
+## Outbox cleanup interval
+
+How outbox cleanup is performed depends per persistence implementation. Most persisters run a cleanup task every minute. Depending on the throughput this cleanup interval can be run more often to make a single cleanup operation purge less records per interval.
```

Comment:
> How outbox cleanup is performed depends per persistence implementation

better as
> Outbox cleanup varies depending on the persistence that is chosen.

Reply:
> Depending on the throughput this cleanup interval can be run more often to make a single cleanup operation purge less records per interval

Depending on the throughput of what? "Depending on the throughput of the endpoints...."? If that's the case, then I'd change it to this:
> Depending on the throughput of the system's endpoints, the Outbox cleanup interval can be run more frequently. This will allow each cleanup operation to purge the fewest records possible each time it is runs, which will allow the cleanup to execute faster.